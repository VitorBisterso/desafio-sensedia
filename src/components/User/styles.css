.user-container {
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: center;
  color: rgba(37, 37, 37, 1);
  justify-content: center;
}

.user-image {
  border-radius: 100%;
  margin-bottom: 10vh;
}

.user-title {
  font-size: 30px;
  font-weight: 800;
}

.user-text {
  font-size: 20px;
}

.user-table td:last-of-type {
  padding-left: 3rem;
}

.user-notFound {
  flex-direction: row;
}

.user-link {
  text-decoration: none;
  color: rgba(69, 36, 187, 1);
  font-weight: bold;
  margin: 0 5px;
}

.user-details {
  display: flex;
  width: 100%;
  flex-direction: row;
  justify-content: space-evenly;
}

.user-details ul {
  margin-top: 3rem;
}

.user-details li {
  margin-top: 1.5rem;
}

@media (max-width: 767px) {
  .user-details {
    flex-direction: column;
    align-items: center;
  }

  .user-details .user-title {
    text-align: center;
  }

  .user-container p:last-of-type {
    text-align: center;
  }
}
