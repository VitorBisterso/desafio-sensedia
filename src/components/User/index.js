import React from 'react';
import { useLocation, Link } from 'react-router-dom';

import './styles.css';

function User() {
  const location = useLocation();
  const userDetails = location.value;

  const renderTableRow = () =>
    Object.keys(userDetails).map(
      item =>
        item !== 'img' && (
          <tr key={item}>
            <td>
              <p className="user-text">{item}:</p>
            </td>
            <td>
              <p className="user-text">{userDetails[item]}</p>
            </td>
          </tr>
        ),
    );

  const renderUserDetails = () => (
    <>
      <img
        className="user-image"
        src={userDetails.img}
        alt="Imagem de perfil do usuário"
      />
      <div className="user-details">
        <div>
          <p className="user-title">Detalhes da conta</p>
          <table className="user-table">
            <tbody>{renderTableRow()}</tbody>
          </table>
        </div>
        <div>
          <p className="user-title">Tecnologias utilizadas</p>
          <ul className="user-text">
            <li>React</li>
            <li>react-router-dom</li>
            <li>react-google-login</li>
          </ul>
        </div>
      </div>
      <p className="user-text">
        Link para o repositório com o código:{' '}
        <a
          className="user-link"
          href="https://github.com/VitorBisterso/desafio-sensedia/"
        >
          https://github.com/VitorBisterso/desafio-sensedia/
        </a>
      </p>
    </>
  );

  const renderEmptyDetails = () => (
    <>
      <p className="user-title">Calma lá, pessoa!</p>
      <p className="user-text">
        Você ainda não está logado com nenhuma conta. Clique
        <Link className="user-link" to="/">
          aqui
        </Link>
        para ser redirecionado à pagina de login.
      </p>
    </>
  );

  const renderContent = () =>
    userDetails ? renderUserDetails() : renderEmptyDetails();

  return <div className="user-container">{renderContent()}</div>;
}

export default User;
