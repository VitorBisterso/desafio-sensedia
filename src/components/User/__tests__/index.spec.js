import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import User from '../index';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/user',
  }),
}));

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <User />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
