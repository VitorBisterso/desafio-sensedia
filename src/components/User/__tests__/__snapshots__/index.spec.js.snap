// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="user-container"
>
  <p
    className="user-title"
  >
    Calma lá, pessoa!
  </p>
  <p
    className="user-text"
  >
    Você ainda não está logado com nenhuma conta. Clique
    <a
      className="user-link"
      href="/"
      onClick={[Function]}
    >
      aqui
    </a>
    para ser redirecionado à pagina de login.
  </p>
</div>
`;
